backoff.go

```diff
@@ -0,0 +1,68 @@
+package deferred
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	maxDelay = 1000 * time.Second
+	factor   = 1.6
+	maxInt64 = float64(math.MaxInt64 - 512)
```

Comment:
The name is slightly confusing to me as I'd expect this to be of type `int64`, should it be called `maxFloat64` instead?